pkg/utils/csr_helpers.go

```diff
@@ -0,0 +1,144 @@
+package utils
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/cloudflare/cfssl/config"
+	"github.com/cloudflare/cfssl/signer"
+	"github.com/cloudflare/cfssl/signer/local"
+	certificatesv1 "k8s.io/api/certificates/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+	"open-cluster-management.io/addon-framework/pkg/agent"
+	addonapiv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+)
+
+// DefaultSigner generates a signer func for addon agent to sign the csr using caKey and caData.
+func DefaultSigner(caKey, caData []byte) agent.CSRSignerFunc {
+	return func(csr *certificatesv1.CertificateSigningRequest) []byte {
+		blockTlsCrt, _ := pem.Decode(caData) // note: the second return value is not error for pem.Decode; it's ok to omit it.
```

Comment:
> // note: the second return value is not error for pem.Decode; it's ok to omit it.
Write Preview

can you elaborate why the error is neglectable? what if the passing caData is not a valid base64 data?

Reply:
``` golang
// Decode will find the next PEM formatted block (certificate, private key
// etc) in the input. It returns that block and the remainder of the input. If
// no PEM data is found, p is nil and the whole of the input is returned in
// rest.
func Decode(data []byte) (p *Block, rest []byte) {
...
```
The second return value is `[]byte` type which we don't need in next steps. And If caData is not a valid base64 data, then the `blockTlsCrt` would be nil.

Reply:
oh i thought its an error